scripts/tunnel-creation.mjs

```diff
@@ -0,0 +1,351 @@
+    // Default port value, will be updated in main() if --packet-stream-base-port is provided
+    this.packetStreamBasePort = 50000;
+    // Default port value, will be updated in main() if --tunnel-registry-port is provided
+    this.tunnelRegistryPort = 42314;
+  }
+
+  /**
+   * Update tunnel registry with new tunnel information
+   * @type {import('appium-ios-remotexpc').TunnelResult[]} results - Array of tunnel results
+   * @returns {Promise<import('appium-ios-remotexpc').TunnelRegistry>} Updated tunnel registry
+   */
+  async updateTunnelRegistry(results) {
+    const now = Date.now();
+    const nowISOString = new Date().toISOString();
+
+    // Initialize registry if it doesn't exist
+    const registry = {
+      tunnels: {},
+      metadata: {
+        lastUpdated: nowISOString,
+        totalTunnels: 0,
+        activeTunnels: 0,
+      },
+    };
+
+    // Update tunnels
+    for (const result of results) {
+      if (result.success) {
+        const udid = result.device.Properties.SerialNumber;
+        registry.tunnels[udid] = {
+          udid,
+          deviceId: result.device.DeviceID,
+          address: result.tunnel.Address,
+          rsdPort: result.tunnel.RsdPort ?? 0,
+          packetStreamPort: result.packetStreamPort,
+          connectionType: result.device.Properties.ConnectionType,
+          productId: result.device.Properties.ProductID,
+          createdAt: registry.tunnels[udid]?.createdAt ?? now,
+          lastUpdated: now,
+        };
+      }
+    }
+
+    // Update metadata
+    registry.metadata = {
+      lastUpdated: nowISOString,
+      totalTunnels: Object.keys(registry.tunnels).length,
+      activeTunnels: Object.keys(registry.tunnels).length, // Assuming all are active for now
+    };
+
+    return registry;
+  }
+
+  /**
+   * Setup cleanup handlers for graceful shutdown
+   */
+  setupCleanupHandlers() {
+    const cleanup = async (signal) => {
+      log.warn(`\nReceived ${signal}. Cleaning up...`);
+
+      // Close all packet stream servers
+      if (this.packetStreamServers.size > 0) {
+        log.info(`Closing ${this.packetStreamServers.size} packet stream server(s)...`);
+        for (const [udid, server] of this.packetStreamServers) {
+          try {
+            await server.stop();
+            log.info(`Closed packet stream server for device ${udid}`);
+          } catch (err) {
+            log.warn(`Failed to close packet stream server for device ${udid}: ${err}`);
+          }
+        }
+        this.packetStreamServers.clear();
+      }
+
+      log.info('Cleanup completed. Exiting...');
+      process.exit(0);
+    };
+
+    // Handle various termination signals
+    process.on('SIGINT', () => cleanup('SIGINT (Ctrl+C)'));
+    process.on('SIGTERM', () => cleanup('SIGTERM'));
+    process.on('SIGHUP', () => cleanup('SIGHUP'));
+
+    // Handle uncaught exceptions and unhandled rejections
+    process.on('uncaughtException', async (error) => {
```

Comment:
if we need to perform the pre-exit cleanup then consider using process.on('exit', ...) hook.
It's more universal, but it only supports synchronous callbacks.
Perhaps, for this purpose we could probably create a synchronous version of the packet server closer